tests/test_parser.py

```diff
@@ -134,38 +119,144 @@ def test_parser_schema_violation():
+                description: Sample extra key
+        ```
+        OrderedDict([('key_1', OrderedDict([('description', 'Sample extra
+        key')])), ('key_2', OrderedDict([('description', 'Sample extra
+        key')])), ('key_3', OrderedDict([('description', 'Sample extra
+        key')])), ('key_4', OrderedDict([('description', 'Sample extra
+        key')])), ('key_5', OrderedDict([('description', 'Sample extra
+        key')])), ('key_6', OrderedDict([('description', 'Sample extra
+        key')])), ('key_7', OrderedDict([('description', 'Sample extra
+        key')])), ('key_8', OrderedDict([('description', 'Sample extra
+        key')])), ('key_9', OrderedDict([('description', 'Sample extra
+        key')])), ('key_10', OrderedDict([('description', 'Sample extra
+        key')])), ('key_11', OrderedDict([('description', 'Sample extra
+        key')]))]) has too many properties
+        Documentation for this node:
+            The acceptable keys on the "extra" object sent with events. This is an
+            object mapping the key to an object containing metadata about the key.
+            A maximum of 10 extra keys is allowed.
+            This metadata object has the following keys:
+                - `description`: **Required.** A description of the key.
+            Valid when `type`_ is `event`.
+        """,
+            ]
+        )
+    else:
+        expected_errors.extend(
+            [
+                """
+        ```
+        gleantest:
+          test_event: !!python/object/apply:collections.OrderedDict
+            - - - type
+                    - event
+        ```
+
+        Missing required properties: bugs, data_reviews, description, expires,
+        notification_emails
+
+        Documentation for this node:
+            Describes a single metric.
+
+            See https://mozilla.github.io/glean_parser/metrics-yaml.html
+        """,
+                """
             ```
             gleantest.event:
-              event_too_many_extras:
-                extra_keys:
-                  key_1:
-                    description: Sample extra key
-                  key_2:
-                    description: Sample extra key
-                  key_3:
-                    description: Sample extra key
-                  key_4:
-                    description: Sample extra key
-                  key_5:
-                    description: Sample extra key
-                  key_6:
-                    description: Sample extra key
-                  key_7:
-                    description: Sample extra key
-                  key_8:
-                    description: Sample extra key
-                  key_9:
-                    description: Sample extra key
-                  key_10:
-                    description: Sample extra key
-                  key_11:
-                    description: Sample extra key
+                event_too_many_extras:
+            extra_keys: !!python/object/apply:collections.OrderedDict
```

Comment:
This funny output (with the `python/object/apply:collections.OrderedDict` thing) is undesirable and confusing -- we want it to look the same as regular YAML input.  You may have to either -- go back to doing something different on Python 3.6, or use the `OrderedDumper` hack currently used only be Python 3.6 on all Python versions.